main.go

```diff
@@ -162,7 +162,14 @@ func PrepareTestSuite(dir string, tc *testCase) (*terraform.Context, *terraspec.
 		return nil, nil, ctxDiags
 	}
 
-	tfCtx, diags := terraspec.NewContext(dir, tc.variableFile, providerResolver) // Setting a different folder works to parse configuration but not the modules :/
+	// Parse specs may return mocked data source result
+	mockMetadata, diags := terraspec.ReadSpecMetadataMock(tc.specFile)
```

Comment:
as you can see I got into a chicken and egg problem 
ReadSpec needs NewContext

ReadSpec is where we are parsing and validating the spec.

Therefore I added ReadSpecMetadataMock (Probably should be ReadSpecMockMetadata) in order to pass it to NewContext where the metadata is set.

Reply:
OK, understood the chicken and egg issue. 
Your current implementation is not ideal as we have to parse the spec file twice. I think we can do better by refactoring the NewContext function.